addons/api/tests/unit/serializers/host-catalog-test.js

```diff
@@ -0,0 +1,108 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+module('Unit | Serializer | host catalog', function (hooks) {
+  setupTest(hooks);
+
+  // Replace this with your real tests.
+  test('it exists', function (assert) {
+    let store = this.owner.lookup('service:store');
+    let serializer = store.serializerFor('host-catalog');
+
+    assert.ok(serializer);
+  });
+
+  test('it serializes records', function (assert) {
+    let store = this.owner.lookup('service:store');
+    let record = store.createRecord('host-catalog', {});
+
+    let serializedRecord = record.serialize();
+
+    assert.ok(serializedRecord);
+  });
+
+  test('it serializes a static host-catalog as expected', async function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const serializer = store.serializerFor('host-catalog');
+    const record = store.createRecord('host-catalog', {
+      name: 'static',
+      description: 'this is a static host-catalog',
+      type: 'static',
+      attributes: {
+        disable_credential_rotation: true,
+        tenant_id: 'a1b2c3',
+      },
+    });
+    const snapshot = record._createSnapshot();
+    const serializedRecord = serializer.serialize(snapshot);
+    const expectedResult = {
+      name: 'static',
+      description: 'this is a static host-catalog',
+      type: 'static',
+    };
+    assert.deepEqual(serializedRecord, expectedResult);
+  });
+
+  test('it serializes an AWS plugin as expected, ignoring read-only and azure fields', async function (assert) {
```

Comment:
We don't need to explicitly exclude or test for the exclusion of type-specific fields.  We only need to test the _inclusion_ of expected fields.  In practice, a model instance should never come to us with fields set for another type.  And even if it did, we want to be somewhat flexible to support unknown types in the future.

I'd modify this test case to test only AWS fields.

Reply:
Gotcha, you are right.
Here I made next (wrong) assumption we want to exclude those fields does not belong to the model or are unexpected as `fake_field`.